proposals/draft/SIPXX - BATCH message type.md

```diff
@@ -0,0 +1,145 @@
+- `time_extracted` **Optional**. The time these records were observed in the source. This should be an RFC3339 formatted date-time, like "2017-11-20T16:45:33.000Z".
+- `size` **Optional**. The number of records per batch file.
+- `format` **Optional**. A JSON map containing serialisation format details. A default json-lines (JSONL) format is expected in the absence of `format` information.
+- `compression` **Optional**. A JSON map containing serialised file compression details.
+- `manifest` **Required**. A list of string URLs indicating the files in this BATCH.
+
+A single Tap may output BATCH messages with different stream
+names.  A single BATCH may only contain records for a single
+stream. The BATCH message may contain pre-signed URLs but it may not directly contain other credential information. Although not strictly required, BATCH message would generally be immediately followed by a STATE message.
+
+Example:
+
+*Note: Every message must be on its own line, but the examples here use multiple lines for readability.*
+
+```json
+{
+  "type": "BATCH",
+  "stream": "stream_name",
+  "time_extracted": "2017-11-20T16:45:33.000Z",
+  "size": 100000, // upper limit rather than file-exact (last file likely to contain fewer)
+  "format": {
+    "name": "csv",
+    "options": {
+      "delimiter": "\t",
+      "quote_char": "\""
+    }
+  },
+  "compression": {
+    "name": null
+    "options": {}
+    "enabled": false
+  },
+  "manifest": [ // can be one or more files
+    "s3://path/to/file001.csv",
+    "s3://path/to/file002.csv",
+    "s3://path/to/file003.csv"
+  ]
+}
+```
+
+## Motivation
+> >
+
+### Which layer(s) of the Singer ecosystem does this proposal directly touch?
+
+Select all that apply:
+
+- [ ] Singer Specification - required capabilities and behaviors
+- [X] Singer Specification - optional capabilities and behaviors
+- [X] Singer best practices and other guidance
+- [ ] Singer Working Group - practices and procedures
+- [X] Singer documentation (Other)
+
+### What problem does it solve?
+
+This proposal addresses 2 different use-cases:
+
+1. Sources with large volumes of upstream records to be retrieved, for example the initial sync of a large (millions of records) RDBMS source.
+1. Sources that support batching natively. This may be warehouses with UNLOAD capabilities or API's with bulk export or reporting endpoints for performantly extracting large volumes of records (vs. per-record or per-page).
+
+This offeres several new avenues for improving performance:
+
+- For taps and targets that _both_ support a common BATCH format (e.g. RDBMS to Warehouse) we have the opportunity to implement 'pass through', whereby records are _never deserialised_ between source to destination, greatly reducing overhead and improving throughput. This would drastically accelerate this common use-case in the singer ecosystem, inspired by the `fast_sync` feature of [Pipelinwise](https://transferwise.github.io/pipelinewise/concept/fastsync.html).
```

Comment:
Perhaps worth mentioning that, especially when dealing with RDBMS, the COPY command is usually the best way to implement this "pass-through. And the COPY command usually requires csv.

Also, and unrelated Pipelinwise -> PipelineWise